src/averageAges.js

```diff
@@ -54,8 +68,23 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
+
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const mothers = people
+    .filter(mother => people.some(child => child.mother === mother.name));
+
+  const children = !onlyWithSon
+    ? people.filter(child => people
+      .some(mother => mother.name === child.mother))
+
+    : people.filter(child => (
+      people.some(mother => mother.name === child.mother) && child.sex === 'm'
+    ));
+
+  return children
+    .map(child => child.born - mothers
+      .find(motherBorn => motherBorn.name === child.mother).born)
```

Comment:
[checklist point 2](https://github.com/mate-academy/js_average-ages/blob/master/checklist.md)

```suggestion
    .map(child => {
       const motherBorn = mothers.find(motherBorn => (
         motherBorn.name === child.mother).born
       ));
      
      return child.born - motherBorn;
    })
```

Reply:
did it cos u smarter and mentor and cos checklist but now it look like spaceship